File: pos_lot_base/static/src/js/models/StockProductionLot.js
/*
    Copyright 2022 Camptocamp SA
    License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
*/
odoo.define("pos_lot_base.StockProductionLot", function (require) {
    "use strict";

    const models = require("point_of_sale.models");

    models.StockProductionLot = window.Backbone.Model.extend({
        initialize: function (attr, options) {
            _.extend(this, options);
        },
    });
});

File: pos_lot_base/models/stock_production_lot.py
# Copyright 2022 Camptocamp SA
# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
from odoo import fields, models


class StockProductionLot(models.Model):

    _inherit = "stock.production.lot"

    product_sale_ok = fields.Boolean(related="product_id.sale_ok")
    product_available_in_pos = fields.Boolean(related="product_id.available_in_pos")
    available_in_pos = fields.Boolean()
